src/core/layertree/qgslayertreelayer.h

```diff
@@ -148,6 +162,8 @@ class CORE_EXPORT QgsLayerTreeLayer : public QgsLayerTreeNode
     QgsMapLayerRef mRef;
     //! Layer name - only used if layer does not exist or if mUseLayerName is false
     QString mLayerName;
+    //! Expression to evaluate in the legend
+    QString mExpression = QString();
```

Comment:
```suggestion
    QString mLabelExpression;
```

The default constructor is used implicitly.

Reply:
done